crates/core/tedge_api/src/entity_store.rs

```diff
@@ -650,6 +656,36 @@ impl EntityTree {
             removed_entities.push(topic_id.clone())
         }
     }
+
+    pub fn list<'a>(&'a self, topic_id: &'a EntityTopicId) -> Vec<&'a EntityMetadata> {
+        if let Some(metadata) = self.entities.get(topic_id).map(|node| node.metadata()) {
+            let mut topic_ids = VecDeque::from(vec![topic_id]);
+            let mut entities = vec![metadata];
+
+            self.list_internal(&mut topic_ids, &mut entities);
```

Comment:
`list_internal` seems over-complicated and doesn't need to be recursive. I would even use a simple `while` loop.

```suggestion
            while let Some(topic_id) = topic_ids.pop_front() {
                let (child_topics, children): (Vec<_>, Vec<_>) =
                    self.children(topic_id).into_iter().unzip();
                topic_ids.extend(child_topics);
                entities.extend(children);
            }
```

Reply:
fixed by https://github.com/thin-edge/thin-edge.io/pull/3355/commits/55975e07897183d946916560587f00d700882735